woocommerce/woocommerce
# Support controlling address card state from data store in Checkout block

This PR moves the state and control of the address card to data store, it means you can know if the card is expanded or not, and you can toggle it off and on via actions.


You can turn an address to card mode if it has invalid fields (unfilled, with errors). I left this at component level via `useEffect` instead of moving it to data store for now.



### Changes proposed in this Pull Request:





Closes https://github.com/woocommerce/woocommerce/issues/48774



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. On an incognito window, visit Checkout.
2. Fill out the address, make sure the address card stays expanded.
3. Fill out a different address for billing, make sure it stays expanded.
4. Place the order.
5. Add an item to cart, go back to Checkout.
6. Make sure both addresses are collapsed.
7. Click edit, make sure you can expand both.
8. Sync up Billing with shipping.
9. Unsync them, make sure billing is open expanded.
10. If you're guest, billing address would be empty, otherwise, it will be filled. (not sure why we separate guest from logged in here, [I asked here](https://github.com/woocommerce/woocommerce/issues/42140#issuecomment-2349106574)).

### Programmatic testing

1. Open Checkout on guest, fill out your shipping address.
2. From console, run `wp.data.dispatch('wc/store/checkout').setEditingShippingAddress(false);`
3. Shipping address should collapse.
4. Run `wp.data.dispatch('wc/store/checkout').setEditingShippingAddress(true);`.
5. Shipping address should expand.
6. Fill out the billing address.
7. Run `wp.data.dispatch('wc/store/checkout').setEditingBillingAddress(false);`
8. It should collapse.
9. Run `wp.data.dispatch('wc/store/checkout').setEditingBillingAddress(true);`
10. Address should expand.
11. Remove fields from billing steps, so that an error is shown to you.
12. Run `wp.data.dispatch('wc/store/checkout').setEditingBillingAddress(false);`
13. Nothing should happen.


### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Move address card state management to data stores in Checkout block.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
